exceptions/app/src/main/cpp/native-lib.cpp

```diff
@@ -4,12 +4,22 @@
 
 #include "exception_helper.h"
 
+void might_throw() { throw std::runtime_error("A C++ runtime_error"); }
+
 extern "C" JNIEXPORT void JNICALL
 Java_com_example_exceptions_MainActivity_throwsException(JNIEnv* env,
                                                          jobject /* this */) {
   try {
-    throw std::runtime_error("A C++ runtime_error");
+    might_throw();
   } catch (std::runtime_error e) {
+    // Re-throw with the error message.
+    jniThrowRuntimeException(env, e.what());
+  } catch (std::exception e) {
+    // A more generic exception. We lose the specific message here.
```

Comment:
We don't actually lose anything, do we? The `std::runtime_error` catch block can be removed without any behavior change, I think.

Reply:
Yeah, you're right.

I just needed to catch by reference because, you know, inheritance is a thing.

Like I said, I don't know how to use exceptions in C++. :)